src/elements/content-uploader/ContentUploader.js

```diff
@@ -628,21 +632,26 @@ class ContentUploader extends Component<Props, State> {
      */
     removeFileFromUploadQueue = (item: UploadItem) => {
         const { onCancel, useUploadsManager } = this.props;
+        const { view, items } = this.state;
         // Clear any error errorCode in footer
         this.setState({ errorCode: '' });
 
         const { api } = item;
         api.cancel();
 
-        const { items } = this.state;
         items.splice(items.indexOf(item), 1);
 
-        // Minimize uploads manager if there are no more items
-        const callback = useUploadsManager && !items.length ? this.minimizeUploadsManager : noop;
-
         onCancel([item]);
-        this.updateViewAndCollection(items, callback);
-        this.upload();
+        this.updateViewAndCollection(items, () => {
+            // Minimize uploads manager if there are no more items
+            if (useUploadsManager && !items.length) {
+                this.minimizeUploadsManager();
+            }
+
+            if (view === VIEW_UPLOAD_IN_PROGRESS) {
```

Comment:
It's not clear to me which state `view` should be referencing since this callback is passed as the callback function to `setState()`

Reply:
See my other comment about the context.